### PR TITLE
feat: first party Deno KV OAuth plugin

### LINKED ISSUE #1733: proposal: add Deno KV OAuth plugin to Fresh
Deno KV OAuth now has a [Fresh plugin](https://github.com/denoland/deno_kv_oauth#getting-started-with-fresh). In [the PR](https://github.com/denoland/deno_kv_oauth/pull/171), there was the idea of having the plugin be located [within the Fresh codebase](https://github.com/denoland/deno_kv_oauth/pull/171#discussion_r1308849938) instead of the Deno KV OAuth codebase. I imagine it looks something like this:
```ts
// main.ts
import { start } from "$fresh/server.ts";
import { createGitHubOAuth2Client, kvOAuthPlugin } from "$fresh/plugins/deno_kv_oauth.ts";
import manifest from "./fresh.gen.ts";

await start(manifest, {
  plugins: [
    kvOAuthPlugin(createGitHubOAuth2Client()),
  ],
});
```
I am for the idea, but my opinion is not strong. Keen to hear other opinions.
CC @jollytoad, @mbhrznr

### PR DESCRIPTION
**Pull request by mbhrznr**
closes #1733.  

migrates the plugin from the kv oauth repository to this one.    
tweaks some minor stuff to align with the other plugins.

### CODE REVIEW

File: plugins/kv_oauth.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+     *
+     * @default {"/oauth/callback"}
+     */
+    callbackPath?: string;
+    /**
+     * Sign-out page path
+     *
+     * @default {"/oauth/signout"}
+     */
+    signOutPath?: string;
```
Comment (iuioiua) [inline plugins/kv_oauth.ts]: Let's move this out into its own `KvOAuthPluginOptions` interface.

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]
LGTM! Thank you for this 👍🏾

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -61,8 +61,8 @@ jobs:
 
       - name: Run tests Windows
         if: matrix.os == 'windows-latest'
-        run: deno test -A --trace-ops
+        run: deno test -A --trace-ops --unstable
```
Comment (marvinhagemeister) [inline .github/workflows/ci.yml:64]: Are the `--unstable` flag needed?

**Review by marvinhagemeister — APPROVED** [review]
This is great! Thanks for making a PR 👍

**Pull request merged**